portal_pages/views.py

```diff
@@ -144,7 +144,7 @@ def submit_blog(request, blog_index):
             send_notification(blog.get_latest_revision().id, 'submitted', None)
         return HttpResponseRedirect(blog_index.url + blog_index.reverse_subpage('thanks'))
 
-    tags = Tag.objects.order_by('name')
+    tags = Tag.objects.filter(id__in=BlogPageTag.objects.all().values('tag')).order_by('name')
```

Comment:
id__in type queries get very expensive as the list of items gets larger. Equivalent and hopefully more efficient as the site grows to have more blog entries with tags would be:

```python
Tag.objects.filter(portal_pages_blogpagetag_items__isnull=False).order_by('name').distinct('name')
```
(This would also be an excellent thing to make sure we add a test for.)